### DISCUSSION TITLE
How to enable other users to see the device screen in monitoring mode?

### DISCUSSION THREAD

**Discussion by merih-sakarya** [DeviceFarmer/stf#652] [General]
Hi there,

I am working on a project where I need to share a device screen in monitoring mode with other users via a shared link. However, I am encountering an issue with the screen socket, as it operates on a session basis and only allows the main controlling user to receive screen image data from the socket.

I understand that this is a security feature, but I am wondering if there is a way to enable other users to receive screen data on DeviceFarmer without compromising security. Do you have any suggestions or workarounds that could help me overcome this issue?"

**Comment by denis99999** [comment]
@merih-sakarya , if you want to share the screen between several users, these users have to login using a shared account, it should works!

**Reply by jupe** [reply]
if actual user is also using shared account, doesn’t it means that “visitor” can mess up actual user session, eg closing it? I guess this was about read-only viewing. I would be also interested this since it would be useful to monitor UI while CI are executing tests..

**Comment by denis99999** [comment]
use a shared account so a unique account on which several users log in, take control of the device or release it is indeed unique by clicking on the button in the '`status`' column, but each user can interact with the device by clicking on the button in the '`product`' column so it is not read-only, and therefore yes indeed you can monitor UI and interact with a device while CI are executing tests.

**Comment by merih-sakarya** [comment]
Thanks for the details you provided. I decided to solve the read-only screen-sharing issue with a separate socket service. Thanks a lot
